File: app/src/main/java/com/example/wangbin/gymclub/MainActivity.java
package com.example.wangbin.gymclub;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView mTextMessage;

    private Fragment1 fragment1;
    private Fragment2 fragment2;
    private Fragment3 fragment3;
    private Fragment4 fragment4;
    private Fragment[] fragments;
    private int lastfragment;


    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    if(lastfragment!=0)
                    {
                        switchFragment(lastfragment,0);
                        lastfragment=0;
                    }

                    return true;
                case R.id.navigation_dashboard:
                    if(lastfragment!=1)
                    {
                        switchFragment(lastfragment,1);
                        lastfragment=1;

                    }
                    return true;
                case R.id.navigation_notifications:
                    if(lastfragment!=2)
                    {
                        switchFragment(lastfragment,2);
                        lastfragment=2;
                    }
                    return true;
                case R.id.navigation_teacher:
                    if(lastfragment!=3)
                    {
                        switchFragment(lastfragment,3);
                        lastfragment=3;

                    }
                    return true;
            }
            return false;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Intent intent = getIntent();
        //获取传递的值
        String str = intent.getStringExtra("name");
        String url=intent.getStringExtra("url");
        fragment1 = new Fragment1();
        fragment2 = new Fragment2();
        fragment3 = new Fragment3();
        fragment4 = new Fragment4();
        Bundle bundle = new Bundle();
        bundle.putString("name",str);
        bundle.putString("url",url);
        fragment1.setArguments(bundle);
        fragments = new Fragment[]{fragment1,fragment2,fragment3,fragment4};
        setContentView(R.layout.activity_main);
        mTextMessage = (TextView) findViewById(R.id.message);
        getSupportFragmentManager().beginTransaction().replace(R.id.mainview,fragment1).show(fragment1).commit();
        BottomNavigationView navigation = (BottomNavigationView) findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);

        lastfragment=0;

    }

    private void switchFragment(int lastfragment,int index)
    {
        FragmentTransaction transaction =getSupportFragmentManager().beginTransaction();
        transaction.hide(fragments[lastfragment]);//隐藏上个Fragment
        if(fragments[index].isAdded()==false)
        {
            transaction.add(R.id.mainview,fragments[index]);
        }
        transaction.show(fragments[index]).commitAllowingStateLoss();

    }


}

File: app/src/main/java/com/example/wangbin/gymclub/MainActivity.kt
package com.example.wangbin.gymclub

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.view.MenuItem
import android.view.ViewParent
import com.example.wangbin.gymclub.adapter.MainFragmentAdapter
import com.example.wangbin.gymclub.fragment.CourseFragment
import com.example.wangbin.gymclub.fragment.HomeFragment
import com.example.wangbin.gymclub.fragment.MyFragment
import com.example.wangbin.gymclub.fragment.TrainerFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(),BottomNavigationView.OnNavigationItemSelectedListener,
        ViewPager.OnPageChangeListener{
    override fun onPageScrollStateChanged(p0: Int) {

    }

    override fun onPageScrolled(p0: Int, p1: Float, p2: Int) {
    }

    override fun onPageSelected(position: Int) {
        main_navigation.menu.findItem(main_navigation.selectedItemId).isChecked = false
        main_navigation.menu.getItem(position).isChecked = true
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when(item.itemId){
            R.id.navigation_home->{
                main_view_pager.setCurrentItem( 0,false)
                return true
            }
            R.id.navigation_dashboard->{
                main_view_pager.setCurrentItem(1,false)
                return true
            }
            R.id.navigation_my-> {
                main_view_pager.setCurrentItem(2,false)
                return true
            }
            R.id.navigation_trainer->{
                main_view_pager.setCurrentItem(3,false)
                return true
            }
        }
        return false
    }

    inline fun <reified T:Activity> Context.startActivity(){
        startActivity(Intent(this,T::class.java))
    }

    inline fun <reified T:Fragment> newInstance():T{
        return T::class.java.newInstance()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        this.supportActionBar?.hide()
        main_navigation.setOnNavigationItemSelectedListener (this)
        var fragmentList:MutableList<Fragment> = ArrayList<Fragment>()
        fragmentList.add(newInstance<HomeFragment>())
        fragmentList.add(newInstance<CourseFragment>())
        fragmentList.add(newInstance<MyFragment>())
        fragmentList.add(newInstance<TrainerFragment>())
        main_view_pager.adapter = MainFragmentAdapter(supportFragmentManager,this,fragmentList)
        main_view_pager.addOnPageChangeListener(this)
    }
}
